patching/src/main/java/org/jboss/as/patching/cli/AbstractDistributionCommand.java

```diff
@@ -296,4 +306,27 @@ private static void formatConflictsList(final StringBuilder buf, final ModelNode
         }
     }
 
+    private void verifyManagementVersion(CLICommandInvocation commandInvocation) throws IOException, CommandException {
+        if (commandInvocation.getCommandContext().isDomainMode()) {
+            // Check the host version to know whether we are allowed to use the command.
+            // Notice the /host=*/core-service=pathing resource won't be available if we do not allow using patching
+            // here we decided to read the host version which is more accurate to do this check.
+
+            ModelControllerClient client = commandInvocation.getCommandContext().getModelControllerClient();
+            commandInvocation.getCommandContext().getModelControllerClient();
+            ModelNode op = new ModelNode();
+            PathAddress address = PathAddress.pathAddress(PathElement.pathElement(HOST, host));
+            op.get(Util.ADDRESS).set(address.toModelNode());
+            op.get(Util.OPERATION).set(ModelDescriptionConstants.QUERY);
+            ModelNode select = new ModelNode().addEmptyList();
+            select.add(MANAGEMENT_MAJOR_VERSION);
+            op.get(SELECT).set(select);
+            ModelNode response = client.execute(op);
+            int major = response.get(RESULT, MANAGEMENT_MAJOR_VERSION).asInt();
+            if (major > 20) {
+                // patch tool unsupported
+                throw new CommandException("'patch' tool is no longer supported on this host controller. For '" + host + "' host you have to use the '" + InstMgrGroupCommand.COMMAND_NAME + "' tool instead.");
```

Comment:
s/'patch' tool/The 'patch' command/g
s/no longer/not/g
s/For '/For the '/g

Reply:
Done